src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Security/BuiltInRole.cs

```diff
@@ -0,0 +1,22 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace OrchardCore.Security
+{
+    public static class BuiltInRole
+    {
+        [Display(Name = nameof(Administrator))]
+        public const string Administrator = "Administrator";
+
+        public const string Anonymous = "Anonymous";
+
+        public const string Authenticated = "Authenticated";
+
+        public const string Author = "Author";
+
+        public const string Contributor = "Contributor";
+
+        public const string Editor = "Editor";
+
+        public const string Moderator = "Moderator";
```

Comment:
Not sure that "Author, Contributor, Editor, Moderator" should be considered built-in roles. Sometimes people want to remove them because they have no use for them.

Reply:
Right, but they are builtin once in OC,. We could add a method say `GetBuiltInRoles()` that can be `virtual`

The PR still WIP, hope to update it frequently ..

Reply:
@Skrypt I tried to go down that path of removing these roles at one point, but there is no way to remove unwanted roles now since all these roles are populated using `IPermissionProvider.GetDefaultStereotypes()` from the IoC container.

One possible way to be able to remove these roles would be to using a service that would provide role types. In `IRole` we can add a property `public string Type { get; set; }`

Then in `IRoleService` we can `Task<IEnumerable<IRole>> GetRolesAsync(string type);`

Then we'll change `IPermissionProvider` from something like this

```
public class UserPermissionProvider : IPermissionProvider
{
    public IEnumerable<PermissionStereotype> GetDefaultStereotypes() =>
        new[]
        {
                new PermissionStereotype
                {
                    Name = "Administrator",
                    Permissions = new[]
                    {
                        UserPermissions.ImportUsingExcel,
                    },
                },
        };

    public Task<IEnumerable<Permission>> GetPermissionsAsync() =>
        Task.FromResult(new[]
        {
                UserPermissions.ImportUsingExcel,
        }
        .AsEnumerable());
}
```

to something like this

```
public class UserPermissionProvider : IPermissionProvider
{
	public IRoleService _roleService;

	public UserPermissionProvider(IRoleService roleService)
	{
		_roleService = roleService
	}

    public IEnumerable<PermissionStereotype> GetDefaultStereotypes()
    {
    	return _roleService.GetRolesAsync("administrator")
    					   .GetAwaiter()
    					   .GetResult()
						   .Select(role => new PermissionStereotype
				           {
				               Name = x.RoleName,
				               Permissions = new[]
				               {
				                   UserPermissions.ImportUsingExcel,
				               },
				           }).ToArray();
    }

    public Task<IEnumerable<Permission>> GetPermissionsAsync() =>
        Task.FromResult(new[]
        {
                UserPermissions.ImportUsingExcel,
        }
        .AsEnumerable());
}
```

Then we would add Delete button to the `Roles` UI.

Reply:
@Skrypt I tried to go down that path of removing these roles at one point, but there is no way to remove unwanted roles now since all these roles are populated using `IPermissionProvider.GetDefaultStereotypes()` from the IoC container.

But I think we should be able to remove default roles. I pitched one way to allow us to delete roles [here](https://github.com/OrchardCMS/OrchardCore/issues/11920#issuecomment-1179769462)

Reply:
Please hold on until the PR is almost ready